src/config/BaseConfig.ts

```diff
@@ -0,0 +1,28 @@
+import ICloudConfig from "../interfaces/Config/ICloudConfig";
+import {INVALID_TYPE_MESSAGE} from "../constants";
+
+class Config {
+  /**
+   * Type guard for user config input
+   * @param {any} object User provided input of unknown shape
+   * @param {string[]} validKeys Array of valid configuration keys
+   * @return {boolean} Whether an object is a valid configuration object
+   */
+  isValidConfig(object:unknown, validKeys:string[]): object is ICloudConfig {
```

Comment:
why `ICloudConfig`? not supposed to be `ICloudinaryConfig`?

Reply:
ICloudinaryConfig is the main Configuration object(that includes cloud, account, url, api etc.)

ICloudConfig is the shape for the CloudConfiguration.